Add kb for editor inside scrollview (#927)
telerik/maui-docs: knowledge-base/dotnet-maui-editor-scroll-inside-scrollview.md

---
title: .NET MAUI Editor inside ScrollView
description: Learn how to scroll the .NET MAUI Editor when in a ScrollView and scrolling the Editor's text when Editor is wrapped in a Border.
type: how-to
page_title: How to Scroll the Editor in a ScrollView - Editor for .NET MAUI
slug: dotnet-maui-editor-scroll-inside-scrollview
tags: editor, scrollview, scrolling, .NET MAUI, multiline entry
res_type: kb
---

## Environment

| Product |
| --- |
| Editor for .NET MAUI |

## Description

> I want the <a href="https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/editor?view=net-maui-8.0" target="_blank">`Editor` for .NET MAUI</a> to scroll on Android when added inside a `ScrollView`.

Reviewer: ```suggestion
I want the <a href="https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/editor?view=net-maui-8.0" target="_blank">Editor for .NET MAUI</a> to scroll on Android when added inside a `ScrollView`.
```

This KB article also answers the following questions:

> - How to use the .NET MAUI `Editor`, so the editor to scroll when having more input elements on the page?

Reviewer: * Don't use the inline code style for the component names.
* Use a capital letter when mentioning the component's name (Editor).
*  Both comments apply to all occurrences of `Editor` and "editor" in this article.
```suggestion
How to use the .NET MAUI Editor, so the Editor scrolls when having more input elements on the page?
```

> - How to use the .NET MAUI `Editor`, so the editor to resize to accommodate new input?

Reviewer: ```suggestion
- How to use the .NET MAUI Editor, so the Editor resizes to accommodate new input?
```

> - How to make the scenarios described above to work correctly when wrapping the .NET MAUI `Editor` inside a Border?

Reviewer: ```suggestion
- How to make the scenarios described above work correctly when wrapping the .NET MAUI Editor inside a border?
```

## Solution

> To scroll the .NET MAUI `Editor` on Android when the editor is inside `ScrollView` you need to use the editor handler to access the native Android control. Then call the  `RequestDisallowInterceptTouchEvent` with parameter `false` to the parent view the editor is placed to. 

Reviewer: ```suggestion
To scroll the .NET MAUI Editor on Android when the Editor is inside a ScrollView:

1. Use the Editor handler to access the native Android control.
1. Call the  `RequestDisallowInterceptTouchEvent` with parameter `false` to the parent view where the Editor is placed.
1. Use the Telerik .NET MAUI [`RadBorder`]({%slug border-overview%}) with a nested Editor to enable the Editor to resize when the content changes.
1. Use the Telerik .NET MAUI [`RadBorder`]({%slug border-overview%}) with a nested Editor to enable the Editor to scroll when the content grows.
```

> Use the Telerik .NET MAUI [RadBorder]({%slug border-overview%}) to achieve the following:

Reviewer: ```suggestion
```

> * `Editor` inside `RadBorder` that resizes when content changes.

Reviewer: Why is this a separate list and doesn't expand on the steps described above?

Author: so the first case is for android, and the second case when using the border is on iOS. We described mainly how to fix issues related in the maui editor using our border control.

> * `Editor` inside `RadBorder` that resizes when content changes.

Reviewer: ```suggestion
```

> * `Editor` inside `RadBorder` that scrolls when content grows.

Reviewer: ```suggestion
```

> ### Example

Reviewer: ```suggestion
```

> Here is an example that solves the questions described above:

Reviewer: ```suggestion
Here is an example that implements the suggested solution:
```

**1.** Define the custom editor in C#:

```C#
public class CustomEditor : Editor
{

}
```

**2.** Define the editor handler:

```C#
public class CustomEditorHandler : EditorHandler
{
#if ANDROID
    protected override AppCompatEditText CreatePlatformView()
    {
        var platfromView = base.CreatePlatformView();
        platfromView.SetOnTouchListener(new CustomTouchListener());
        return platfromView;
    }

    class CustomTouchListener : Java.Lang.Object, IOnTouchListener
    {
        public bool OnTouch(Android.Views.View? v, MotionEvent? e)
        {
            v?.Parent?.RequestDisallowInterceptTouchEvent(true);
            if ((e.Action & MotionEventActions.Up) != 0 &&
                (e.ActionMasked & MotionEventActions.Up) != 0)
            {
                v?.Parent?.RequestDisallowInterceptTouchEvent(false);
            }

            return false;
        }
    }
#endif
}

```

> **3.** To resize the editor to accommodate new inputs set its `AutoSize` property to `TextChanges`. 

Reviewer: ```suggestion
**3.** To resize the Editor to accommodate new input, set its `AutoSize` property to `TextChanges`. 
```
